extensions/workspace/watch/watcher.ts

```diff
@@ -129,7 +135,8 @@ export class Watcher {
     }
     const idStr = componentId.toString();
     const hook = isChange ? 'OnComponentChange' : 'OnComponentAdd';
-    logger.console(`running ${hook} hook for ${chalk.bold(idStr)}`);
+
+    this.pubsub.pub(WorkspaceAspect.id, this.creatOnComponentChangeEvent(idStr, hook));
```

Comment:
- [ ] do not remove the logger console, if it break the console, change it to logger.debug
- [ ] add another event when the hook is OnComponentAdd
- [ ] add another event for component remove inside `executeWatchOperationsOnRemove`